drivers: mcux: imx rt flexcan support

rtalbott opened the pull request:
Adds flexcan support to the imx rt series.

MaureenHelm commented on drivers/can/can_mcux_flexcan.c:
```diff
@@ -14,6 +14,10 @@
 
+#if defined(CONFIG_NET_SOCKETS_CAN)
+#include "socket_can_generic.h"
+#endif /* CONFIG_NET_SOCKETS_CAN */
```
How is this change related to imx rt?

rtalbott replied:
this is part of an existing feature. I moved it back to the socket can area.

MaureenHelm commented on drivers/can/can_mcux_flexcan.c:
```diff
@@ -587,17 +591,17 @@ static void mcux_flexcan_transfer_callback(CAN_Type *base,
 	switch (status) {
 	case kStatus_FLEXCAN_UnHandled:
-		__fallthrough;
+		/* fallthrough */
```
why are you changing these fallthroughs? is it a rebasing error?

rtalbott replied:
yes it was.. fixed now

MaureenHelm reviewed (changes requested):
Thanks for the contribution @rtalbott . 

Could you please enable this on at least one board? Please add `can` to the board supported list, similar to `boards/arm/frdm_k64f/frdm_k64f.yaml`

henrikbrixandersen reviewed (changes requested):
You need to change the commit message to include a body text in addition to the headline.

rtalbott committed:
drivers: mcux: imx rt flexcan support

Adds imx rt support.
Allows n-number of can interfaces based on device-tree.
Adds a "common" irq name.
Added CAN bus pins and dts for 1060 and 1064 EVK.

Signed-off-by: Rick Talbott <rtalbott@fastmail.com>

henrikbrixandersen commented on dts/arm/nxp/nxp_rt.dtsi:
```diff
@@ -620,6 +620,57 @@
+			phase-seg1 = <3>;
+			phase-seg2 = <2>;
+			status = "disabled";
+			#address-cells = <1>;
+			#size-cells = <0>;
```
Are these needed?

henrikbrixandersen replied:
Nevermind me, they are.

henrikbrixandersen reviewed (changes requested):
Can you set a default for `CAN_MAX_FILTER` in `drivers/can/Kconfig.mcux` when running on i.MX?

henrikbrixandersen reviewed (approved):
LGTM
